### ISSUE TITLE
Generated SKILL.md install command uses the importable module name, not the PyPI distribution name

### ISSUE THREAD

**Issue by mikeckennedy** [posit-dev/great-docs#272]
## Summary

For packages whose **importable module name differs from their PyPI distribution name**, the generated Agent Skill (`SKILL.md` / `skill.md`, the `.well-known` copies, `skills.html`, and `search.json`) emits an install command built from the **module name** rather than the **distribution name**. The result is an install command that installs the wrong package (or nothing related).

Concrete case: [`umami-analytics`](https://pypi.org/project/umami-analytics/) is imported as `umami`. The generated skill produces:

```bash
pip install umami
```

but `pip install umami` pulls an **unrelated** project from PyPI. The correct command is `pip install umami-analytics` (the distribution name).

- Version: great-docs 0.15.0
- Live example (note the "Install this skill" / Installation command): https://mkennedy.codes/docs/umami-python/skills.html
- The published skill: https://mkennedy.codes/docs/umami-python/skill.md

## Why this happens

The skill generator resolves the install name from the **importable module name** (via the quarto `api-reference.package` value, which must be the importable name for introspection), e.g. in `core.py`:

```python
install_name = package_name.replace("_", "-") if package_name else ""
...
lines.append(f"pip install {install_name}")
```

Interestingly, `_detect_package_name()` already reads the correct **distribution** name from `pyproject.toml` (`project.name` -> `umami-analytics`); it just isn't the value used for the skill's install line. So the right name is already available — it's a matter of using the distribution name for the *install command* while keeping the module name for *introspection*.

There is also no config knob to override just the install/distribution name for the skill (the skill config exposes `file`, `well_known`, `gotchas`, `best_practices`, `decision_table`, `extra_body`, `skills` — none of which set the install name; `pypi:` controls the PyPI link, not the command).

## Suggested fix

- Use the detected **distribution name** (`project.name` from `pyproject.toml`, normalized) for the generated `pip install ...` line, rather than the importable module name.
- When `pypi:` is set to a string (custom index) or you want an explicit override, honor that for the install command too.
- Optionally, add a small skill-config override (e.g. `skill.install_name` or `skill.distribution`) for cases where metadata can't be detected.

## Impact

This is a correctness issue for any project where import name ≠ dist name (common: `umami-analytics`/`umami`, `beautifulsoup4`/`bs4`, `pillow`/`PIL`, `scikit-learn`/`sklearn`, etc.). Agents that follow the generated skill's install command will install the wrong package.

## Workaround

Until this is fixed upstream, I added a post-build step to my docs pipeline that rewrites the install command in the generated artifacts (`skill.md`, the `.well-known` `SKILL.md` copies, `skills.html`, and `search.json`) to `uv pip install umami-analytics`, using exact anchored substrings so it doesn't touch `import umami` or `umami-analytics`. Happy to share the snippet if useful.

**Comment by mikeckennedy** [comment]
Thanks for adding this @rich-iannone I've been converting a ton of my projects to Great Docs:

https://mkennedy.codes/docs/

I've hit a couple of issues. I'll try to keep track. This was the biggest one that was generating fully wrong instructions.

**Comment by rich-iannone** [comment]
@mikeckennedy thanks for your continued support! Any time you might run into something off please do create an issue (we are in a phase of work that's mostly about fixing what's already there).
